build_swagger_spec.py

```diff
@@ -9,19 +9,40 @@
 
 parser = argparse.ArgumentParser()
 parser.add_argument('app', help='the flask app to swaggerify')
-#parser.add_argument('--definitions', help='json definitions file')
-parser.add_argument('--template', help='template spec to start with')
+parser.add_argument('--template', help='template spec to start with, before any other options or processing')
 parser.add_argument('--out-dir', default=None, help='the directory to output to')
+parser.add_argument('--definitions', default=None, help='json definitions file')
+parser.add_argument('--host', default=None)
+parser.add_argument('--base-path', default=None)
+parser.add_argument('--version', default=None, help='Specify a spec version')
+
 args = parser.parse_args()
 
 def run():
-    template = args.template
     app = pkg_resources.EntryPoint.parse("x=%s" % args.app).load(False)
-    if template is not None:
-        with open(template, 'r') as f:
-            spec = swagger(app, template=json.loads(f.read()))
-    else:
-        spec = swagger(app)
+
+    # load the base template
+    template = None
+    if args.template is not None:
+        with open(args.template, 'r') as f:
+            template = json.loads(f.read())
+
+    # overwrite template with specified arguments
+    if args.definitions is not None:
```

Comment:
All the conditionals below assume `template` to be a dict and not None.
If someone provides `host` and not `template` the script will fail with an exception.
Either move the conditionals under a check for `template is not None` or provide handling for invalid arguments